### PR TITLE
FEAT: Ability to reorder & ability to hide Cameras in UI

### PR DESCRIPTION
**Pull request by NickM-27**
Implementation of #2782
Implementation of #2837 

• Add GUI config (for other modifications to the GUI in the future)
• Add order to GUI config
• Sort cameras by order. The default is 0 for every camera so the current order is maintained in the default case.

### CODE REVIEW

**Comment by NickM-27** [comment]
@yury-sannikov per your comment about `key` on the previous PR, can you help me understand what it is used for. Would I use the camera object as the key?

**Comment by blakeblackshear** [comment]
https://reactjs.org/docs/lists-and-keys.html

**Comment by NickM-27** [comment]
@blakeblackshear Thanks! Updated with keys on cameras and sidebar items that this PR touches. Will keep that in mind in any future PRs as well.

**Comment by NickM-27** [comment]
@blakeblackshear Where is the mock config for the tests stored that swr loads? For this PR to work I need to add a couple variables to the test config and I can't find it 🤔

**Comment by blakeblackshear** [comment]
@NickM-27 you mock the actual API call here: https://github.com/blakeblackshear/frigate/blob/release-0.11.0/web/config/handlers.js

File: docker-compose.yml [hunk 1]
```diff
@@ -5,16 +5,16 @@ services:
...
+      - "44"  # video
+      - "46"  # plugdev
     shm_size: "256mb"
     build:
       context: .
       dockerfile: docker/Dockerfile.dev
-    devices:
-      - /dev/bus/usb:/dev/bus/usb
-      - /dev/dri:/dev/dri # for intel hwaccel, needs to be updated for your hardware
+    #devices:
```
Comment (blakeblackshear) [inline docker-compose.yml:15]: Did you have to comment these out for it to work? Are you running the latest docker and docker-compose versions?

File: docs/docs/configuration/index.md [hunk 2]
```diff
@@ -431,4 +431,12 @@ cameras:
       quality: 70
       # Optional: Restrict mqtt messages to objects that entered any of the listed zones (default: no required zones)
       required_zones: []
+
+    # Optional: Configuration for how camera is handled in the GUI.
+    gui:
```
Comment (blakeblackshear) [inline docs/docs/configuration/index.md:436]: Let's use `ui` here since we already have a global `ui` entry.

File: docker-compose.yml (hunk 1 above)
Comment (NickM-27) [reply docker-compose.yml:15]: Oh sorry didn't mean to include that in the PR. But yeah I did have to, I am running Docker Hub on Mac so I will check for update and see.

File: web/src/Sidebar.jsx [hunk 3]
```diff
@@ -21,9 +21,12 @@ export default function Sidebar() {
           matches ? (
             <Fragment>
               <Separator />
-              {Object.entries(cameras).map(([camera]) => (
-                <Destination key={camera} href={`/cameras/${camera}`} text={camera} />
-              ))}
+              {Object.entries(cameras)
```
Comment (blakeblackshear) [inline web/src/Sidebar.jsx:24]: Probably best to do this in a `useMemo` above so this logic doesn't run on every render.

Comment (NickM-27) [reply web/src/Sidebar.jsx:24]: I am having a difficult time figuring out the useMemo. I think I have the memo logic figured out but it says that a memo can not be used in a conditional and only be used in the same order every time. I am assuming it is referring to the `match` condition. I looked it up and seems like I may need to split the sorted cameras part into its own function, do you have any recommendations on a clean way to do that?

Comment (blakeblackshear) [reply web/src/Sidebar.jsx:24]: You have to call `useMemo` above the null config check in the component.

Comment (NickM-27) [reply web/src/Sidebar.jsx:24]: Nevermind I think I finally got it

Comment (NickM-27) [reply web/src/Sidebar.jsx]: Okay, let me know if what I did works. Also should I do the same in the `Cameras.jsx` as well?

Comment (blakeblackshear) [reply web/src/Sidebar.jsx]: Yes.

Comment (NickM-27) [reply web/src/Sidebar.jsx]: Cool, seems to be working well now after my latest commits 👍

**Comment by NickM-27** [comment]
Going to close and reopen this due to the fragmented commit history.

**Pull request closed without merging**